File: 520_DetectCapital.py
#82.51%
class Solution(object):
    def detectCapitalUse(self, word):
        """
        :type word: str
        :rtype: bool
        """
        cnt = 0
        for c in word:
            if c.isupper():
                cnt += 1
        return cnt == 0 or cnt == len(word) or cnt == 1 and word[0].isupper()
		
	def detectCapitalUse(self, word):
		return word.isupper() or word.islower() or word.istitle()

File: 520_DetectCapital.cpp
//17.96%
class Solution {
public:
    bool detectCapitalUse(string word) {
        if(word.empty())
			return true;
		bool firstChr = word[0] >= 'A' && word[0] <= 'Z' ? true : false;
		bool secondChr(false);
		//Other chars need to be all upper or lower alphabets while firstChr is upper
		int cnt(0);
		for(int i = 1; i < word.size(); ++i){
			if(firstChr){
				if(word[i] >= 'A' && word[i] <= 'Z')
					++cnt;
			}else{
				if(word[i] >= 'A' && word[i] <= 'Z')
					return false;
			}
		}
		return cnt == 0 || cnt == word.size() - 1;		
    }
};

//3.93%
class Solution {
public:
    bool detectCapitalUse(string word) {
		int cnt(0);
		for(char c : word){
			if(c >= 'A' && c <= 'Z')
				++cnt;
		}
		return cnt == 0 || cnt == word.size() || cnt == 1 && word[0] >= 'A' && word[0] <= 'Z';		
    }
};

//Java by using regex
public boolean detectCapitalUse(String word) {
    return word.matches("[A-Z]+|[a-z]+|[A-Z][a-z]+");
}



